src/decisionengine/framework/taskmanager/TaskManager.py

```diff
@@ -145,10 +141,15 @@ def __init__(self, name, generation_id, channel_dict, global_config):
         self.id = str(uuid.uuid4()).upper()
         self.dataspace = dataspace.DataSpace(global_config)
         self.data_block_t0 = datablock.DataBlock(self.dataspace, name, self.id, generation_id)  # my current data block
-        self.name = name
-        self.logger = structlog.getLogger(f"{self.name}")
-        self.logger = self.logger.bind(module=__name__.split(".")[-1])
-        self.channel = Channel(channel_dict, name)
+
+        if "channel_name" in channel_dict.keys():
+            self.name = channel_dict["channel_name"]
```

Comment:
Can I get a unit test for this `if` condition?

Reply:
Done!